app/utils/queryBuilder.ts

```diff
@@ -416,4 +427,24 @@ export const useQueryBuilder = (
 ): string =>
   useMemo(() => {
     return buildQuery(queryConfig, contextFilters);
+
   }, [queryConfig, contextFilters]);
+
+/**
+ * Hook which builds a graphql query for record exporting
+ */
+export const useExportQuery = <T extends Record<string, unknown>>(
+  queryConfig: QueryConfig,
+  contextFilters?: Filter[]
+): string => {
+  const newQueryConfig = useMemo(() => {
+    // update the provided query with export settings
+    return produce(queryConfig, (newQueryConfig) => {
+      // reset limit and offset
+      newQueryConfig.limit = MAX_RECORD_EXPORT_LIMIT;
```

Comment:
Hasura will accept a `limit` of `null`, but modifying this typing to accept `null` was a headache i didn't really want to deal with 🤷